### DISCUSSION TITLE
Reg Artillery & Tekton CI/CD pipeline integration guide

### DISCUSSION THREAD

**Discussion by rrnareshkumar** [artilleryio/artillery#2111] [Artillery]
Hi , If we have guide with information on how to set up & trigger Artillery from Tekton CI/CD pipelines, Please share the link.

**Comment by bernardobridge** [comment]
Hello @rrnareshkumar. We don't have a specific guide for Tekton CI/CD pipelines, but we do have a section in our Documentation for setting up Artillery with CI/CD pipelines and a few popular examples (https://www.artillery.io/docs/cicd) which you can use as inspiration. 

In the case of `Tekton`, I believe it is container-based, so you should be able to use Artillery's official container image (`artilleryio/artillery:latest`), and then run an Artillery script using that image.

**Reply by rrnareshkumar** [reply]
thanks for the updates , I will try it out .

**Reply by rrnareshkumar** [reply]
Hi , I can see the Artillery's official container image By [artilleryio](https://hub.docker.com/u/artilleryio) • Updated 19 days ago .

Can you let us know what this docker image contains , for example : dependencies (such as node.js and its version , npm and its version) and any other files etc

if there is doc or link available (about this docker image) for more detailed information about build/run/test instructions , please share .

**Reply by bernardobridge** [reply]
Hi @rrnareshkumar . The official Docker image is https://hub.docker.com/r/artilleryio/artillery.

The Dockerfile for that image is here: https://github.com/artilleryio/artillery/blob/main/packages/artillery/Dockerfile . The information about node version and what files are included can be obtained from there.

**Reply by rrnareshkumar** [reply]
Hi , thanks , this is helpful . 

I noticed that Artillery requires the most recent LTS release of Node.js which is 18.x (and npm version in 9.x) but in my organization approved node.js version to install on containers and use is  17.x (and its corresponding npm version is 8.x) so i cant use the most recent LTS release of Node.js.

In this case , May i know if it possible for me to install artillery.io using node.js version 17.x (and its corresponding npm is 8.x) and use the artillery tool for load testing without any issues ? Please suggest

**Reply by bernardobridge** [reply]
Hi @rrnareshkumar. That's not something we support out of the box, as 17.x already reached `End of Life` last year - also, older versions like 16.x will also be `EOL` next week (https://github.com/nodejs/release#release-schedule), so a lot of organisations will need to move to 18.x+ anyway.

But, if you would like to use Artillery with older versions of node, then you would have to install it yourself, using a base docker image in Tekton that has that node version, and then manually installing Artillery in that image.

_Note: we don't guarantee that newer versions of Artillery will work with older versions of node. We only support 18.x and 20.x at the moment._

---

Alternatively, you can try to use an older docker image version (which will also use an older Artillery version), but I don't think we have published versions using version 17 of node anyway, only 18, 16 or 14.
